src/npc.h

```diff
@@ -118,7 +118,7 @@ class Npc final : public Creature
 		}
 
 		bool isPushable() const final {
-			return walkTicks > 0;
+			return walkTicks > 0 && pushable;
```

Comment:
Test for pushability first, `pushable && walkTicks > 0`. Actually, since `walkTicks` is unsigned, you could test `!= 0` instead.